analyses/npl/04_glofas_flood_events.md

```diff
@@ -0,0 +1,255 @@
+                
+            FP = sum(forecast_detections == 0)
+            df_station_stats = df_station_stats.append({
+                'station': station,
+                'leadtime': leadtime,
+                'rp': rp,
+                'TP': TP,
+                'FP': FP,
+                'FN': FN,
+            }, ignore_index=True)
+
+```
+
+```python
+df_station_stats['precision'] = df_station_stats['TP'].astype(int) / (df_station_stats['TP'].astype(int) + df_station_stats['FP'].astype(int))
+df_station_stats['recall'] = df_station_stats['TP'].astype(int) / (df_station_stats['TP'].astype(int) + df_station_stats['FN'].astype(int))
+```
+
+```python
+#rp_dict = {
+#    1.5: '-', 
+#    2: '--',
+#    5: ':'
+#}
+rp_dict = {MAIN_RP: '-'}
+plot_numbers = True
+plot_precision_recall = True
+leadtime_range = (1, 10)
+
+for istation, station in enumerate(FINAL_STATIONS):
+    if plot_numbers:
+        qs = ['TP', 'FP', 'FN']
+        fig, ax = plt.subplots()
+        for rp, ls in rp_dict.items():
+            data = df_station_stats[(df_station_stats['station'] == station) & (df_station_stats['rp'] == MAIN_RP)]
+            for iq, q in enumerate(qs):
+                ax.plot(data['leadtime'], data[q], ls=ls, c=f'C{iq}')
+        ax.set_title(station)
+        ax.xaxis.set_major_locator(MaxNLocator(integer=True))
+        ax.yaxis.set_major_locator(MaxNLocator(integer=True))
+        ax.set_xlabel('Leadtime [days]')
+        ax.set_ylabel('Number')
+        # Make legend
+        for iq, q in enumerate(qs):
+            ax.plot([], [], c=f'C{iq}', label=q)
+        if len(rp_dict) > 1:
+            for rp, ls in rp_dict.items():
+                ax.plot([], [], c='k', ls=ls, label=rp)
+        ax.legend()
+        ax.set_xlim(leadtime_range)
+    
+    if plot_precision_recall:
+        qs = ['precision', 'recall']
+        fig, ax = plt.subplots()
+        for rp, ls in rp_dict.items():
+            data = df_station_stats[(df_station_stats['station'] == station) & (df_station_stats['rp'] == MAIN_RP)]
+            for iq, q in enumerate(qs):
+                ax.plot(data['leadtime'], data[q], ls=ls, c=f'C{iq}')
+        ax.set_title(station)
+        ax.xaxis.set_major_locator(MaxNLocator(integer=True))
+        ax.set_xlabel('Leadtime [days]')
+        ax.set_ylabel('Fraction')
+        ax.set_ylim(-0.1, 1.1)
+        # Make legend
+        for iq, q in enumerate(qs):
+            ax.plot([], [], c=f'C{iq}', label=q)
+        if len(rp_dict) > 1:
+            for rp, ls in rp_dict.items():
+                ax.plot([], [], c='k', ls=ls, label=rp)
+        ax.legend()
+        ax.set_xlim(leadtime_range)
+```
+
+```python
+# Print out some stats
+def round_to_5(x):
+    return (np.around(x/5, decimals=0)*5).astype(int)
+for station in FINAL_STATIONS:
+    data = df_station_stats[(df_station_stats['station'] == station) & (df_station_stats['rp'] == rp)]
+    data.loc[:, "POD"] = round_to_5(data["recall"].fillna(-1) * 100)
```

Comment:
Why round to 5?

Reply:
I included these numbers in a presentation, and rounding to 10 seemed a bit too coarse. I've added a comment to explain.